app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -225,23 +225,46 @@ def deploy_agent
     security_group_id = find_or_create_security_group(vpc_id)
     find_or_create_profile
 
-    instance = ec2.create_instances(
-      :iam_instance_profile => {:name => label},
-      :image_id             => get_agent_image_id,
-      :instance_type        => 't2.micro',
-      :key_name             => kp.name,
-      :max_count            => 1,
-      :min_count            => 1,
-      :placement            => {:availability_zone => zone_name},
-      :tag_specifications   => [{:resource_type => "instance", :tags => [{:key => "Name", :value => label}]}],
-      :network_interfaces   => [{
-        :associate_public_ip_address => true,
-        :delete_on_termination       => true,
-        :device_index                => 0,
-        :subnet_id                   => subnet_id,
-        :groups                      => [security_group_id]
-      }],
-    ).first
+    # Based on Amazon doc, add a retry logic in creating instance to solve time issue on IAM role.
+    #
+    # Important
+    #
+    # After you create an IAM role, it may take several seconds for the permissions to propagate.
+    # If your first attempt to launch an instance with a role fails, wait a few seconds before trying again.
+    # For more information, see Troubleshooting Working with Roles in the IAM User Guide.
+    #
+    # (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#launch-instance-with-role-console)
+
+    max_retries = 3
```

Comment:
You may want to add some more retries, 20 seconds may not be enough.

Reply:
Changed retry time to 5, totally wait for 60 seconds.